test/e2e/resize_nodes.go

```diff
@@ -230,109 +270,213 @@ var _ = Describe("ResizeNodes", func() {
 	})
 
-	testName = "should be able to delete nodes."
-	It(testName, func() {
-		Logf("starting test %s", testName)
-
-		if !providerIs(supportedProviders...) {
-			By(fmt.Sprintf("Skipping %s test, which is only supported for providers %v (not %s)",
-				testName, supportedProviders, testContext.Provider))
-			return
-		}
-
-		if testContext.CloudConfig.NumNodes < 2 {
-			By(fmt.Sprintf("skipping %s test, which requires at lease 2 nodes (not %d)",
-				testName, testContext.CloudConfig.NumNodes))
-			return
-		}
-
-		// Create a replication controller for a service that serves its hostname.
-		// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
-		name := "my-hostname-delete-node-" + string(util.NewUUID())
-		replicas := testContext.CloudConfig.NumNodes
-		createServeHostnameReplicationController(c, ns, name, replicas)
-		pods, err := waitForPodsCreated(c, ns, name, replicas)
-		Expect(err).NotTo(HaveOccurred())
-		e := waitForPodsRunning(c, pods)
-		if len(e) > 0 {
-			Failf("Failed to wait for pods running: %v", e)
-		}
-		err = verifyPodsResponding(c, ns, name, pods)
-		Expect(err).NotTo(HaveOccurred())
-
-		By(fmt.Sprintf("decreasing cluster size to %d", replicas-1))
-		err = resizeNodeInstanceGroup(replicas - 1)
-		Expect(err).NotTo(HaveOccurred())
-		err = waitForNodeInstanceGroupSize(replicas - 1)
-		Expect(err).NotTo(HaveOccurred())
-		err = waitForClusterSize(c, replicas-1)
-		Expect(err).NotTo(HaveOccurred())
-
-		By("verifying whether the pods from the removed node are recreated")
-		pods, err = waitForPodsCreated(c, ns, name, replicas)
-		Expect(err).NotTo(HaveOccurred())
-		e = waitForPodsRunning(c, pods)
-		if len(e) > 0 {
-			Failf("Failed to wait for pods running: %v", e)
-		}
-		err = verifyPodsResponding(c, ns, name, pods)
-		Expect(err).NotTo(HaveOccurred())
+	Describe("Resize", func() {
+		AfterEach(func() {
+			if !providerIs(supportedProviders...) {
+				return
+			}
+			By("restoring the original node instance group size")
+			if err := resizeNodeInstanceGroup(testContext.CloudConfig.NumNodes); err != nil {
+				Failf("Couldn't restore the original node instance group size: %v", err)
+			}
+			if err := waitForNodeInstanceGroupSize(testContext.CloudConfig.NumNodes); err != nil {
+				Failf("Couldn't restore the original node instance group size: %v", err)
+			}
+			if err := waitForClusterSize(c, testContext.CloudConfig.NumNodes); err != nil {
+				Failf("Couldn't restore the original cluster size: %v", err)
+			}
+		})
+
+		testName = "should be able to delete nodes."
+		It(testName, func() {
+			Logf("starting test %s", testName)
+
+			if !providerIs(supportedProviders...) {
+				By(fmt.Sprintf("Skipping %s test, which is only supported for providers %v (not %s)",
+					testName, supportedProviders, testContext.Provider))
+				return
+			}
+
+			if testContext.CloudConfig.NumNodes < 2 {
+				By(fmt.Sprintf("skipping %s test, which requires at lease 2 nodes (not %d)",
```

Comment:
As above.  Definitely fail here, as we will almost never have a test cluster explicitly sized to less than 2 nodes.